tests/drivers/sent/sent_api/src/main.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sent/sent.h>
+
+#define SENT_NODE          DT_ALIAS(sent_node)
+#define SENT_CHANNEL       1
+#define SENT_MAX_RX_BUFFER 1
+
+const struct device *sent_dev = DEVICE_DT_GET(SENT_NODE);
+
+struct sent_frame serial_frame[SENT_MAX_RX_BUFFER];
+
+struct sent_frame fast_frame[SENT_MAX_RX_BUFFER];
+
+static void *sent_setup(void)
+{
+	zassert_true(device_is_ready(sent_dev), "SENT device is not ready");
+
+	return NULL;
+}
+
+void rx_serial_frame_cb(const struct device *dev, uint8_t channel_id, uint32_t num_frame,
+			void *user_data)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+	ARG_UNUSED(num_frame);
+	ARG_UNUSED(user_data);
+}
+
+void rx_fast_frame_cb(const struct device *dev, uint8_t channel_id, uint32_t num_frame,
+		      void *user_data)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+	ARG_UNUSED(num_frame);
+	ARG_UNUSED(user_data);
+}
+
+/**
+ * @brief Test starting rx is not allowed while started.
+ */
+ZTEST_USER(sent_api, test_start_listening_while_started)
+{
+	int err;
+
+	err = sent_start_listening(sent_dev, SENT_CHANNEL);
+	zassert_equal(err, 0, "Failed to start rx (err %d)", err);
+
+	err = sent_start_listening(sent_dev, SENT_CHANNEL);
+	zassert_not_equal(err, 0, "Started rx while started");
```

Comment:
I would suggest using z_assert_ok() / z_assert_not_ok() instead for all these kinds of asserts